src/XrdHttp/XrdHttpReq.cc

```diff
@@ -858,11 +805,10 @@ void XrdHttpReq::parseResource(char *res) {
   resourceplusopaque = resource;
   // Whatever comes after is opaque data to be parsed
   if (strlen(p) > 1) {
-    buf = unquote(p + 1);
-    opaque = new XrdOucEnv(buf);
+    std::string decoded = decode_str(p + 1);
+    opaque = new XrdOucEnv(decoded.c_str());
     resourceplusopaque.append('?');
     resourceplusopaque.append(p + 1);
```

Comment:
Is this correct? Shouldn't `opaque` be used to append?

Reply:
Nice catch!

Reply:
NOPE :D

Reply:
The resourceplusopaque string should contain the encoded string (I checked out master and did the following breakpoint):
```
(gdb) n
862	    opaque = new XrdOucEnv(buf);
(gdb) p buf
$4 = 0x7fffe8012e10 "test=test["
(gdb) n
863	    resourceplusopaque.append('?');
(gdb) n
250507 10:56:04 2371660 daemon.2451780:28@ccaffy-eosdev-al9 Xrootd_Protocol: f702 request timeout; read 0 of 24 bytes
250507 10:56:04 2371660 Xrd_Poll: Poller 0 enabled daemon.2451780:28@ccaffy-eosdev-al9
864	    resourceplusopaque.append(p + 1);
(gdb) n
865	    free(buf);
(gdb) p resourceplusopaque
$5 = {_vptr.XrdOucString = 0x41cc10 <vtable for XrdOucString+16>, str = 0x7fffe8004f20 "/tmp/fic.txt_dest?test=test%5B", len = 30, siz = 31, 
  static blksize = -1}
(gdb) 
```